src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs

```diff
@@ -149,5 +149,49 @@ public void TFS_470021()
                 Assert.Equal(0, errorCount);
             }
         }
+
+        /// <summary>
+        /// Test for issue #30218, resource Sch_MinLengthGtBaseMinLength
+        /// </summary>
+        [Fact]
+        public void MinLengthGtBaseMinLength_Throws()
+        {
+            string schema = @"<?xml version='1.0' encoding='utf-8' ?>
+<xs:schema elementFormDefault='qualified'
+           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
+    <xs:simpleType name='foo'>
+        <xs:restriction base='xs:string'>
+            <xs:minLength value='5'/>
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:simpleType name='bar'>
+        <xs:restriction base='foo'>
+            <xs:minLength value='4'/>
+        </xs:restriction>
+    </xs:simpleType>
+</xs:schema>
+";
+
+            XmlSchemaSet ss = new XmlSchemaSet();
+            ss.Add(null, XmlReader.Create(new StringReader(schema)));
+
+            Exception exception;
+
+            try
+            {
+                ss.Compile();
+                exception = null;
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
+            }
+
+            Assert.NotNull(exception);
+            Assert.IsType<XmlSchemaException>(exception);
+            Assert.Equal("It is an error if the derived 'minLength' facet value is less than the parent 'minLength' facet value.",
```

Comment:
We do not currently localize strings, but if we did so in the future (which would certainly be optional if we did) we would need tests to pass on the localized builds, and this would not.

You can either just assert for the string to contain "minLength" -- so it's at least not some random unrelated string. Or I think it would be OK to not add a test at all, especially if you believe the functional code coverage already exists.

Reply:
I am having difficulty checking the code coverage.  I can't even get most of the tests to run.  I'm pretty sure I'm doing something wrong.  Maybe you can advise me.

Here is how I'm building the runtime.  (I'm using git bash as the command line.)
`$ git clean -xdf`
`$ git fetch upstream master`
`$ git merge upstream/master`
`$ git push origin master`
`$ ./build.cmd -subsetCategory coreclr-runtime -c Checked`
`$ ./build.cmd -subsetCategory coreclr-libraries -runtimeConfiguration Checked`
`$ ./build.cmd -vs System.Private.Xml`

In Visual Studio, I press Ctrl + Shift + B to build the solution.  Only 15 of 29 projects build successfully.

The first error in the output window is:

> NuGet package restore failed. Please see Error List window for detailed warnings and errors.

However, there is no additional information in the Output window.  Additionally, building at the command line does not experience this problem.

The test projects which do not build all show this error:

> error NU1201: Project XmlCoreTest is not compatible with netcoreapp5.0 (.NETCoreApp,Version=v5.0). Project XmlCoreTest supports: net50 (.NETFramework,Version=v5.0)

I'm using Community Edition, so I'll have to check code coverage via command line testing anyway, so I tried this:
`cd src/libraries/System.Private.Xml`
`../../../.dotnet/dotnet test -p:CollectCoverage=true -p:CoverletOutputFormat=cobertura`

This results in numerous errors like this:

> Starting test execution, please wait...
>
>A total of 1 test files matched the specified pattern.
>Testhost process exited with error: It was not possible to find any compatible framework >version
>The framework 'Microsoft.NETCore.App', version '5.0.0' was not found.
 > - The following frameworks were found:
 >    2.1.11 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.12 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
 >     2.1.16 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      3.0.0-preview9-19423-09 at [C:\Program  Files\dotnet\shared\Microsoft.NETCore.App]
>      3.0.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      3.1.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
>      5.0.0-preview.3.20167.12 at >[C:\Users\mrj00\source\repos\runtime\.dotnet\shared\Microsoft.NETCore.App]
>You can resolve the problem by installing the specified framework and/or SDK.
>The specified framework can be found at:
>  - https://aka.ms/dotnet-core-applaunch?>framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=x64&rid=win10-x64
>. Please check the diagnostic logs for more information.
>
>Test Run Aborted.

This appears to be the same error.  Any ideas what I'm doing wrong?  No coverlet.cobertura.xml file is then found anywhere in the project folders.

Reply:
Forgot to mention: I have DOTNET_ROOT set to the .NET core in the runtime/.dotnet folder.

However, when I run this from the System.Private.Xml folder
`../../../.dotnet/dotnet --version`

It still prints out 3.1.200.  I don't understand why as this is the local copy of .NET core downloaded by the build script.

Reply:
I've made some progress.
I've set DOTNET_MULTILEVEL_LOOKUP to 0, and started invoking it through the dotnet.cmd  script.

Running the tests now produces many errors like the following.  

> Starting test execution, please wait...
>
>A total of 1 test files matched the specified pattern.
>Testhost process exited with error: It was not possible to find any compatible framework version
>The framework 'Microsoft.NETCore.App', version '5.0.0' was not found.
>  - The following frameworks were found:
>      5.0.0-preview.3.20167.12 at [C:\Users\mrj00\source\repos\runtime\.dotnet\shared\Microsoft.NETCore.App]
>You can resolve the problem by installing the specified framework and/or SDK.
>The specified framework can be found at:
>  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=5.0.0&arch=x64&rid=win10-x64
. Please check the diagnostic logs for more information.
>
>Test Run Aborted.

Where are the "diagnostic logs"?  Why is this framework not acceptable for version '5.0.0'?  It is the internally bundled framework, so I expect it should be acceptable.